app/src/main/java/com/babylon/wallet/android/presentation/transaction/analysis/GeneralTransferAnalysis.kt

```diff
@@ -122,32 +122,57 @@ private fun TransactionType.GeneralTransaction.resolveFromAccounts(
     }
 }
 
+/**
+ * The account deposits order that comes from RET does not take into account the order we maintain in the app.
+ * This method sorts accounts in the same order they are appearing in the dashboard.
+ */
+fun Map<String, List<ResourceTracker>>.sort(allAccounts: List<Network.Account>): Map<String, List<ResourceTracker>> {
+    val allAccountsAddresses = allAccounts.map { it.address }
+
+    // Only account deposits that we own
+    val ownedAccountDeposits = this.toList().filter {
+        allAccountsAddresses.indexOf(it.first) != -1
+    }
+
+    // Sorted owned accounts deposits according to the all accounts order
+    val ownedAccountDepositsSorted = ownedAccountDeposits.sortedBy {
+        allAccountsAddresses.indexOf(it.first)
+    }.toMap()
+
+    // The rest of account deposits (ones we do not own)
+    val thirdPartyAccountDeposits = this.minus(ownedAccountDepositsSorted.keys)
+
+    return ownedAccountDepositsSorted.plus(thirdPartyAccountDeposits)
+}
+
```

Comment:
You can also use the `Map<String, List<ResourceTracker>>.toSortedMap(comparator)` where you can create your own comparator according to the current List<Network.Account> and avoid the complex logic.

Reply:
> You can also use the `Map<String, List<ResourceTracker>>.toSortedMap(comparator)` where you can create your own comparator according to the current List<Network.Account> and avoid the complex logic. This sorts the keys according to the comparator which is what you want.

Yeah I came up with something like this :
```
toSortedMap(
        Comparator.comparing { item -> allAccounts.map { it.address }.indexOf(item) }
    )
```
But that doesn't serve the purpose cause negatives are at the front (third party accounts).
I will keep it as is to speed up release and revisit it at later time.